### ISSUE TITLE
fedora 39 input chinese is garbled code

### ISSUE THREAD

**Issue by stockxiay** [neovide/neovide#2224]
fedora 39 input  chinese then is garbled code
<img width="404" alt="截屏2023-12-30 22 56 34" src="https://github.com/neovide/neovide/assets/17241275/f2d12397-76d3-43b0-bda1-f164536bd24a">

**Comment by fredizzimo** [comment]
I think we need a bit more information, what’s your `guifont` setting? And a logfile from `neovide —log, without any sensitive data, everything that you type and show inside neovide is logged.

**Comment by fredizzimo** [comment]
For the guifont, at least noto cjk works well as a fallback font, see 
https://neovide.dev/configuration.html#font for the configuration.

**Comment by stockxiay** [comment]
if vim.g.neovide then
    -- font
    vim.api.nvim_set_option_value("guifont", "FiraMono Nerd Font Mono:h14:b", {})

    -- padding

**Comment by fredizzimo** [comment]
Yes, you need to add a fallback font. The automatic system fallbacks, do not work that reliably, and in some cases also need to be propely set up on the system. I have some changes in progress that will improve it a little bit but, it’s still recommended to add a fallback font that supports Chinese, such as https://github.com/notofonts/noto-cjk

**Comment by xpbluesmile** [comment]
I have installed the font, but I am using the X11 platform and it still cannot be displayed correctly.
![2](https://github.com/neovide/neovide/assets/3887637/f3b3fa51-efaa-4ab9-b5cd-63ab8869875f)
![2024-03-19_16-55](https://github.com/neovide/neovide/assets/3887637/366ec8bd-b7b4-4399-9c31-db2a9189a4b0)

**Comment by xpbluesmile** [comment]
I removed the ‘google-noto-sans-cjk-vf-fonts’ and related dependencies , then installed ‘google-noto-sans-cjk-fonts’ and back to the Chinese environment. everything worked fine.

**Comment by ueaner** [comment]
Set Chinese font in the `~/.config/neovide/config.toml`.

See: https://github.com/ueaner/dotfiles/commit/bf5fb5d7f6afac59ef50b5dae47f73128d352206

```toml
[font]
size = 12

[font.normal]
family = "Noto Sans Mono CJK SC"
style = "Regular"
```

Works fine under Fedora 40.
